cvm/docker-compose.yml

```diff
@@ -22,12 +22,38 @@ services:
       --async-scheduling
     restart: unless-stopped
     healthcheck:
-      start_interval: 1h30m # Allow up to 90 minutes for initial model loading
+      start_period: 1h30m # Allow up to 90 minutes for initial model loading
       test: ["CMD", "curl", "-f", "http://localhost:8000/health"]
       interval: 30s
       timeout: 10s
       retries: 3
 
+  proxy_api:
+      build:
+        context: ./vllm_proxy
+        dockerfile: Dockerfile
+      container_name: proxy_api
+      depends_on:
+        vllm:
+          condition: service_started
+          restart: true
+      networks:
+        - vllm
+      environment:
+        - BASE_VLLM_URL=http://vllm:8000
+      # No external ports - internal access only
+      expose:
+        - "7000"
+      command: >
+        uv run uvicorn proxy_app:app --host 0.0.0.0 --port 7000
```

Comment:
You can avoid using `uvicorn` (that's what FastAPI suggest) and use `uv run fastapi run ...` instead, which is the official FastAPI runner. I also think this should go into the Dockerfile as the image should start the server by default (e.g. Add an entrypoint.sh in `vllm_proxy/` and use it inside the Dockerfile `ENTRYPOINT ["/app/entrypoint.sh"]`).